### PR TITLE
add service save in etcd

### PR DESCRIPTION
**Pull request by freignat91**
Relates to #162 

This PR add services data in ETCD
## Verification

```
$ ./swarm start
$ make test-stack
```

### CODE REVIEW

File: api/rpc/stack/stack_services.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Server) saveServiceInETCD(ctx context.Context, stackID string, service *ServiceConfig) error {
+	data, _ := json.Marshal(service)
+	val := &storage.Service{Name: service.Name, Data: string(data)}
+	err := s.Store.Create(ctx, stackRootKey+"/"+stackID+servicesRootKey+"/"+service.Name, val, nil, 0)
```
Comment (subfuzion) [inline api/rpc/stack/stack_services.go]: Nitpicking, but please use [path.Join](https://golang.org/pkg/path/#Join) instead.

**Review by subfuzion — CHANGES_REQUESTED** [review]
Looks good, just one nitpicky suggestion.

File: api/rpc/stack/stack.go [hunk 2]
```diff
@@ -34,9 +19,13 @@ func (s *Server) Up(ctx context.Context, in *UpRequest) (*UpReply, error) {
 	}
 	stackID := stringid.GenerateNonCryptoID()
 	stack.StackId = stackID
-	s.Store.Create(ctx, "stacks/"+stackID, stack, nil, 0)
+	s.Store.Create(ctx, stackRootKey+"/"+stackID, stack, nil, 0)
```
Comment (subfuzion) [inline api/rpc/stack/stack.go]: Same comment  about using `path.Join`...

File: api/rpc/stack/stack_services.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+			if err := s.saveServiceInETCD(ctx, stackID, service); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (s *Server) saveServiceInETCD(ctx context.Context, stackID string, service *ServiceConfig) error {
+	data, _ := json.Marshal(service)
```
Comment (generalhenry) [inline api/rpc/stack/stack_services.go]: why json?

**Review by generalhenry — CHANGES_REQUESTED** [review]

File: api/rpc/stack/stack_services.go (hunk 1 above)
Comment (freignat91) [reply api/rpc/stack/stack_services.go]: ok done

File: api/rpc/stack/stack.go (hunk 2 above)
Comment (freignat91) [reply api/rpc/stack/stack.go]: done too

File: api/rpc/stack/stack_services.go (hunk 3 above)
Comment (freignat91) [reply api/rpc/stack/stack_services.go]: right, i'm going to use the proto message directly

**Comment by freignat91** [comment]
Waiting for the last Tony's service proto service message to update the PR accordlingly

**Comment by subfuzion** [comment]
#177 has been merged.

File: api/rpc/stack/stack_services.go (hunk 3 above)
Comment (subfuzion) [reply api/rpc/stack/stack_services.go]: Yep, only protobufs https://github.com/appcelerator/amp/blob/master/data/storage/interface.go#L24

**Review by generalhenry — APPROVED** [review]

File: api/rpc/stack/stack_services.go (hunk 3 above)
Comment (freignat91) [reply api/rpc/stack/stack_services.go]: done

**Comment by freignat91** [comment]
Waiting for new parsing using service.ServiceSpec instead of stack.Service

**Comment by subfuzion** [comment]
@freignat91 Closing this now -- go ahead and reopen it when you're ready.

**Pull request closed without merging**